### PR TITLE
feature: Display institution information on book homepage

### PR DESCRIPTION
**Pull request by fdalcin**
Partial fix for pressbooks/pressbooks-aldine#305

This PR adds the [newly created](https://github.com/pressbooks/pressbooks/pull/2587) Institutions info to the book homepage.

#### How to test

1. Add one or more institutions to the book in the Book Info option
2. Go to the book homepage and check if the information is displayed properly (when the book has no institutions, this won't be shown in the homepage)

### CODE REVIEW

File: inc/helpers/namespace.php [hunk 1]
```diff
@@ -442,6 +442,7 @@ function get_metakeys() {
 		'pb_book_license' => __( 'License', 'pressbooks-book' ),
 		'pb_primary_subject' => __( 'Primary Subject', 'pressbooks-book' ),
 		'pb_additional_subjects' => __( 'Additional Subject(s)', 'pressbooks-book' ),
+		'pb_institutions' => __( 'Institution(s)', 'pressbooks-book' ),
```
Comment (SteelWagstaff) [inline inc/helpers/namespace.php:445]:
```suggestion
'pb_institutions' => _n_noop( 'Institution', 'Institutions', 'pressbooks-book' ),
```

Comment (SteelWagstaff) [reply inc/helpers/namespace.php:445]: Maybe something like this?

File: partials/content-cover-metadata.php [hunk 2]
```diff
@@ -33,6 +33,8 @@
 									$output[] = \Pressbooks\Metadata\get_subject_from_thema( $code );
 								}
 								$book_information[ $key ] = implode( ', ', $output );
+							} elseif ( 'pb_institutions' === $key ) {
+								$book_information[ $key ] = implode( ', ', $book_information[ $key ] );
```
Comment (SteelWagstaff) [inline partials/content-cover-metadata.php:37]: Not sure how to write the case here to display the correct singular/plural form, but I think the solution involves `translate_nooped_plural`: https://developer.wordpress.org/reference/functions/translate_nooped_plural/

**Review by SteelWagstaff — CHANGES_REQUESTED** [review]
I'd like us to display `Institution` if only one institution has been selected and `Institutions` if more than one has been selected rather than `Institutions(s)` in either case. I'm not sure if the AccessLint nag needs attention, but we should look at that as well?

File: inc/helpers/namespace.php (hunk 1 above)
Comment (fdalcin) [reply inc/helpers/namespace.php:445]: It's a first step, but requires some more changes. I'll work on it.

File: inc/helpers/namespace.php [hunk 3]
```diff
@@ -442,6 +442,7 @@ function get_metakeys() {
 		'pb_book_license' => __( 'License', 'pressbooks-book' ),
 		'pb_primary_subject' => __( 'Primary Subject', 'pressbooks-book' ),
 		'pb_additional_subjects' => __( 'Additional Subject(s)', 'pressbooks-book' ),
+		'pb_institutions' => _n_noop( 'Institution', 'Institutions', 'pressbooks-book' ),
```
Comment (fdalcin) [inline inc/helpers/namespace.php:445]: Will render `Institution` or `Institutions` based on the number of institutions.

File: inc/helpers/namespace.php [hunk 4]
```diff
@@ -605,12 +606,14 @@ function comments_template( $comment, $args, $depth ) {
...
  *
  * @return int
  */
-function count_authors( $var ) {
-	return count( \Pressbooks\Utility\oxford_comma_explode( $var ) );
+function count_items( $value ): int {
+	$countable = is_countable( $value ) ? $value : \Pressbooks\Utility\oxford_comma_explode( $value );
+
+	return count( $countable );
 }
```
Comment (fdalcin) [inline inc/helpers/namespace.php:617]: Renamed the method since it no longer counts only authors.

**Review by SteelWagstaff — APPROVED** [review]
LGTM

**Pull request merged**